rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1750,10 +1702,9 @@ extern "C"
             {
                 void *data = guard_conditions->guard_conditions[i];
                 GuardCondition *guard_condition = (GuardCondition*)data;
-                if(!guard_condition->getHasTriggered())
-                {
-                    guard_conditions->guard_conditions[i] = 0;
-                }
+                // Guard conditions persist (don't set to null)
+                // We call getHasTriggered to reset the trigger state.
```

Comment:
Yeah, that would be great, but unfortunately we don't have console logging yet :stuck_out_tongue:. And this is the kind of thing you want wrapped in `#ifndef NDEBUG` and on a "super debug level" so it doesn't print all the time. Just printing to stderr probably wouldn't be appropriate in this case. Otherwise I would have just suggested we implement the debugging now.

It shouldn't be difficult (and shouldn't take much effort) to re-add this functionality in the future if/when we have logging and need to debug it. So based on that and you're opinion, I'd say it's safe to remove it.